packages/react-instantsearch-hooks-server/src/getServerState.tsx

```diff
@@ -8,19 +8,21 @@ import {
 import type { InitialResults, InstantSearch, UiState } from 'instantsearch.js';
 import type { IndexWidget } from 'instantsearch.js/es/widgets/index/index';
 import type { ReactNode } from 'react';
-import type { renderToString as reactRenderToString } from 'react-dom/server';
 import type {
   InstantSearchServerContextApi,
   InstantSearchServerState,
 } from 'react-instantsearch-hooks';
 
 type SearchRef = { current: InstantSearch | undefined };
 
+export type RenderToString = (element: JSX.Element) => unknown;
+
 /**
  * Returns the InstantSearch server state from a component.
  */
 export function getServerState(
-  children: ReactNode
+  children: ReactNode,
+  renderToString?: RenderToString
```

Comment:
Should we add a comment to make it mandatory next major version?

Reply:
should there also be a warning if it's not given (knowing that it does work in most cases)?

Reply:
I was not for it initially, but maybe it's best to warn early since it will be a quick adjustment on those who update the library, and will remove the issues outlined in the description.